delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resolvers/ResolverResolution.scala

```diff
@@ -177,11 +188,12 @@ final class ResolverResolution[R](
         // No resolution was successful yet, we carry on
         case (f: ResolverFailedReport, _)  =>
           val resolve = for {
-            _        <- validateIdentities(projectRef)
-            resource <- fetch(ref, projectRef).flatMap { res =>
-                          IO.fromOption(res)(ResolutionFetchRejection(ref, projectRef))
-                        }
-            _        <- validateResourceTypes(extractTypes(resource), projectRef)
+            _            <- validateIdentities(projectRef)
+            resourceOpt  <- fetch(ref, projectRef)
+            resource     <- IO.fromOption(resourceOpt)(ResolutionFetchRejection(ref, projectRef))
```

Comment:
I preferred this previously. An option would be to make a method to hide the boilerplate